conda_build/skeletons/pypi.py

```diff
@@ -313,17 +302,15 @@ def skeletonize(packages, output_dir=".", version=None, recursive=False,
             base_url = urlsplit(pypi_url)
             base_url = "://".join((base_url.scheme, base_url.netloc))
             ordered_recipe['source']['url'] = urljoin(base_url, ordered_recipe['source']['url'])
+            ordered_recipe['source']['sha256'] = d['hash_value']
 
             if d['entry_points']:
                 ordered_recipe['build']['entry_points'] = d['entry_points']
 
             if noarch_python:
                 ordered_recipe['build']['noarch'] = 'python'
 
-            ordered_recipe['build']['script'] = 'python setup.py install ' + ' '.join(setup_options)
-            if any(re.match(r'^setuptools(?:\s|$)', req) for req in d['build_depends']):
-                ordered_recipe['build']['script'] += ('--single-version-externally-managed '
-                                                      '--record=record.txt')
+            ordered_recipe['build']['script'] = 'python -m pip install . --no-deps --ignore-installed'
```

Comment:
If we are forcing pip by default, we also need to disable the hermetic build stuff, since that duplicates conda-build's intent and can also break what conda-build does.

https://github.com/pypa/pip/pull/5053/files#diff-1457c307ea972916e3ade69c27f0c41aR456

Now the thing that really stinks is that that flag is only available on new pip, so it'll break old builds, or ecosystems that don't have pip 10 yet.

I'm OK with the rest of the changes in this PR, but I don't think we're ready for this one yet.

Reply:
I think there is a environment variable we could set which would turn off build isolation and still work with pip 9

Reply:
great, that would be better.  I'll see if we can find it.

Reply:
Can we add `--no-cache-dir` to the pip arguments as well (to force it to re-create/re-download the wheel and not rely on local cache, if present)? I was bit by this quite a few times.

Reply:
`PIP_NO_BUILD_ISOLATION=False` disables build isolation.  My expectation was that this should be set to True but pypa/pip#5229 and my own testing confirms that False is the correct setting.  This variable has no effect on older pip versions.

I would also suggest adding `-vvv` to increase verbosity which makes debugging failed builds much easier.